File: cpp/Harddrive.cpp
// Harddrive.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "HardwareKomponen.cpp"

using namespace std;

class Harddrive : public virtual Komponen, public HardwareKomponen
{
private:
    int kapasitasGB;
    string tipeDrive;
public:
    Harddrive() {}

    Harddrive(int kapasitasGB, string tipeDrive, string merk, string nama, string model, int konsumsiDayaWatt)
        : Komponen(merk, nama), HardwareKomponen(model, konsumsiDayaWatt), kapasitasGB(kapasitasGB), tipeDrive(tipeDrive) {}

    void setKapasitasGB(int kapasitasGB) { this->kapasitasGB = kapasitasGB; }
    void setTipeDrive(string tipeDrive) { this->tipeDrive = tipeDrive; }

    int getKapasitasGB() const { return kapasitasGB; }
    string getTipeDrive() const { return tipeDrive; }

    ~Harddrive() {}
};

File: py/Harddrive.py
from Komponen import Komponen
from HardwareKomponen import HardwareKomponen


# Harddrive.py
class Harddrive(Komponen, HardwareKomponen):
    def __init__(self, kapasitasGB=0, tipeDrive="", merk="", nama="", model="", konsumsiDayaWatt=0):
        Komponen.__init__(self, merk, nama)
        HardwareKomponen.__init__(self, model, konsumsiDayaWatt)
        self.kapasitasGB = kapasitasGB
        self.tipeDrive = tipeDrive
    
    def setKapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB
    
    def setTipeDrive(self, tipeDrive):
        self.tipeDrive = tipeDrive
    
    def getKapasitasGB(self):
        return self.kapasitasGB
    
    def getTipeDrive(self):
        return self.tipeDrive
